src/vm/docs/mod.rs

```diff
@@ -497,16 +523,26 @@ has to be a literal function name.",
 (fold - (list 3 7 11) 2) ;; Returns 5 
 (define-private (concat-string (a (string-ascii 20)) (b (string-ascii 20))) (unwrap-panic (as-max-len? (concat a b) u20)))
 (fold concat-string \"cdef\" \"ab\")   ;; Returns \"fedcab\"
-(fold concat-string (list \"cd\" \"ef\") \"ab\")   ;; Returns \"efcdab\"",
+(fold concat-string (list \"cd\" \"ef\") \"ab\")   ;; Returns \"efcdab\"
+(define-private (concat-buff (a (buff 20)) (b (buff 20))) (unwrap-panic (as-max-len? (concat a b) u20)))
+(fold concat-buff 0x03040506 0x0102)   ;; Returns 0x060504030102
+",
 };
 
 const CONCAT_API: SpecialAPI = SpecialAPI {
-    input_type: "(buff, buff)|(list, list)",
-    output_type: "buff|list",
-    signature: "(concat buff-a buff-b)",
-    description: "The `concat` function takes two buffers or two lists with the same entry type,
-and returns a concatenated buffer or list of the same entry type, with max_len = max_len_a + max_len_b.",
-    example: "(concat \"hello \" \"world\") ;; Returns \"hello world\""
+    input_type: "
+buff, buff
```

Comment:
Seconded.  I see that `|` is used in other places in this diff to represent a choice between types.  Can the documentation be consistent?

Reply:
The '\n instead of |' was an idea that @kantai and I came up with to avoid overly long lines. The idea was that '\n' would have the same semantics a '|'. But, there are definitely a confusing number of variations.

Also, I just tested how some of these changes look using the 'docs' repo. It seems that the html generator doesn't respect the newline anyway. It seems the line just runs on, so this probably isn't a workable idea.

https://paste.pics/fe67b8ecfe29e5ba945f9728f5b6b349

On the other hand, even if we put the '|' in the line, it will still be really hard to read, if the line still just runs on.

@jcnelson how would you want to handle this?

What do you think of, e.g., the following for `fold`?

`Function((buff 1)|A|(string-ascii 1)|(string-utf8 1), B) -> B, buff|(list A)|string-ascii|string-utf8, B`

To me, it seems a little confusing, but this might be the most consistent way to handle this case.

Reply:
I updated the docs to use a kind of "virtual type", where a sequence of type "X" is said to have type "sequence_X". 

Then, each description using such a sequence type has the quote
"Applicable sequence types are `(list A)`, `buff`, `string-ascii` and `string-utf8`,
for which the corresponding element types are, repsectively, `A`, `(buff 1)`, `(string-ascii 1)` and `(string-utf8 1)`."

Reply:
To me, it is much more readable now with the `sequence` virtual type.